pkg/collector/corechecks/cluster/kubernetes_state_transformers.go

```diff
@@ -0,0 +1,62 @@
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// metricTransformerFunc is used to tweak or generate new metrics from a given KSM metric
+// For name translation only please use metricNamesMapper instead
+type metricTransformerFunc = func(aggregator.Sender, string, ksmstore.DDMetric, []string)
+
+var (
+	// metricTransformers contains KSM metric names and their corresponding transformer functions
+	// These metrics require more than a name translation to generate Datadog metrics, as opposed to the metrics in metricNamesMapper
+	// TODO: implement the metric transformers of these metrics and unit test them
+	// For reference see METRIC_TRANSFORMERS in KSM check V1
+	metricTransformers = map[string]metricTransformerFunc{
+		"kube_pod_status_phase":                       func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_pod_container_status_waiting_reason":    func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_pod_container_status_terminated_reason": func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_cronjob_next_schedule_time":             func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_job_complete":                           func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_job_failed":                             func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_job_status_failed":                      func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_job_status_succeeded":                   func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_node_status_condition":                  func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_node_spec_unschedulable":                func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_resourcequota":                          resourcequotaTransformer,
+		"kube_limitrange":                             func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_persistentvolume_status_phase":          func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+		"kube_service_spec_type":                      func(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {},
+	}
+)
+
+// resourcequotaTransformer generates dedicated metrics per resource per type from the kube_resourcequota metric
+func resourcequotaTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, tags []string) {
+	resource, found := metric.Labels["resource"]
+	if !found {
+		log.Debugf("Couldn't find 'resource' label, ignoring metric '%s'", name)
+		return
+	}
+	typ, found := metric.Labels["type"]
```

Comment:
could we use quotaType or something instead of typ ?

Reply:
yes sure 👍